win32_event_log/datadog_checks/win32_event_log/check.py

```diff
@@ -387,10 +402,12 @@ def create_subscription(self):
             Bookmark=self._bookmark_handle if bookmark else None,
         )
 
-        # https://docs.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtcreaterendercontext
-        # https://docs.microsoft.com/en-us/windows/win32/api/winevt/ne-winevt-evt_render_context_flags
-        self._render_context_system = win32evtlog.EvtCreateRenderContext(win32evtlog.EvtRenderContextSystem)
-        self._render_context_data = win32evtlog.EvtCreateRenderContext(win32evtlog.EvtRenderContextUser)
+    def reset_subscription(self):
+        # Destroy old subscription
+        # Important so that it doesn't affect event or bookmark handle states.
+        self._subscription = None
```

Comment:
Do you think it still worth to call [EvtClose](https://docs.microsoft.com/en-us/windows/desktop/api/winevt/nf-winevt-evtclose). Even if the handle is bad just in case?

Reply:
Did you check that the current bookmark survived/persisted

Reply:
`EvgClose` will be called automatically.

http://timgolden.me.uk/pywin32-docs/PyEVT_HANDLE.html
> When the object is destroyed, EvtClose is called.

The bookmark does survive. In the test I added the event is reported before `reset_subscription` is called, the bookmark is the reason the new subscription sees the prior event and the test passes.